backend/pkg/handler/channels.go

```
@@ -65,6 +75,10 @@ func (h *Handler) CreateChannel(ctx echo.Context, appID string) error {
 }
 
 func (h *Handler) GetChannel(ctx echo.Context, appID string, channelID string) error {
+	appID, err := h.db.GetAppID(appID)
+	if err != nil {
+		return appNotFoundResponse(ctx, appID)
+	}
```

We aren't using the appID here, AFAICS, maybe we don't need the extra query?

---

I moved the check to handler function because of two reasons, 
1. There are handler functions where the appId is used in more than one `api` functions like `PaginateChannels`, So it made sense to find the appID once and reuse it.
2. If the `appID` is not found for the `productID` the response will be a `App not found error` instead of Internal Server Error, we can bubble up the error but keeping it this way makes it more clear both at handler level and api level, At handler level it helps for better error handling and at api level the functions do one specific thing.

---

Ill remove these extra ones.